subfun/design/setSpeedTargets.m

```diff
@@ -0,0 +1,19 @@
+% [ W I P ]
+
+function speeds = setSpeedTargets(cfg)
+    % Compute the matrix with the speed targets if requested, otherwise output will be only zeros
+    %
+    % (C) Copyright 2021 CPP visual motion localizer developpers
+
+    if ismember('speed', cfg.target.type)
+        [~, nbEventsPerBlock, ~, nbBlocks] = getDesignInput(cfg);
+        speeds = ones(nbBlocks, nbEventsPerBlock) * cfg.dot.speedPixPerFrame;
+
+        % Outputs an "empty" matrix in case no speed task is required
+    else
+        [~, nbEventsPerBlock, ~, nbBlocks] = getDesignInput(cfg);
+        speeds = ones(nbBlocks, nbEventsPerBlock) * cfg.dot.speedPixPerFrame;
+
+    end
```

Comment:
I know what confused me on this function.

At the moment both sides of the if block do exactly the same thing, so I thought either the If block is not necessary or the else part must be doing something different from the rest.

Do you see what I mean ?

Reply:
Yes you are totally right, kind of impossible to guess. My bad.

Let's say that now the function has the structure for future developments. Let me add some "warnings" here and there to make sure it is clear

Reply:
just a couple of words below the [WIP] will do. :-)